src/resty/aws/credentials/ChainableTemporaryCredentials.lua

```diff
@@ -104,21 +104,30 @@ end
 -- updates credentials.
 -- @return success, or nil+err
 function ChainedTemporaryCredentials:refresh()
-  local result, err = self.sts:assumeRole(self.params)
-  if not result then
-    return nil, err
+  local response, err = self.sts:assumeRole(self.params)
+  if not response then
+    return nil, "Request for token data failed: " .. tostring(err)
   end
 
-  if type(result) == "string" then
-    local res, err = cjson.decode(result)
-    if not res then
-      return nil, "failed to json-decode assumeRole results with '" .. err .. "', input: " .. result
-    end
-    result = res
+  if response.status ~= 200 then
+    return nil, ("request for token returned '%s': %s"):format(tostring(response.status), response.body)
```

Comment:
shouldn't `response.body` also be wrapped in a `tostring`?

Reply:
I think it is not needed after I add back the `response.body` type check, what do you think about it?